Helper/PaymentMethods.php

```diff
@@ -133,41 +138,51 @@ public function getPaymentMethods($quoteId, $country = null)
     {
         // get quote from quoteId
         $quote = $this->quoteRepository->getActive($quoteId);
-        $store = $quote->getStore();
 
-        $paymentMethods = $this->fetchAlternativeMethods($store, $country);
+        // If quote cannot be found early return the empty paymentMethods array
+        if (empty($quote)) {
+            return [];
+        }
+
+        $this->setQuote($quote);
```

Comment:
why adding a setQuote() method as we can easily do $this->quote = $quote ?

Reply:
There was already a getter created and by generally talking getters and setters are good to use because you can have better control over your property. In the current case we don't get these benefits because the $quote, the getQuote() and setQuote() are all protected but can be useful for later.